src/coreclr/jit/morph.cpp

```diff
@@ -2129,7 +2129,6 @@ void CallArgs::AddFinalArgsAndDetermineABIInfo(Compiler* comp, GenTreeCall* call
             if (arg.NewAbiInfo.HasAnyFloatingRegisterSegment())
             {
                 // Struct passed according to hardware floating-point calling convention
-                assert(arg.NewAbiInfo.NumSegments <= 2);
```

Comment:
For my understanding... what changed here? In what cases do these take more segments to describe now?

Reply:
Ah, I guess the assert is just subsumed by the code+assert below.

Reply:
Nothing, this assert was redundant. `if (arg.NewAbiInfo.NumSegments == 2) {...} else { assert(arg.NewAbiInfo.NumSegments == 1); }` is stricter because it excludes 0.